### PR TITLE
fix: update regex to account for bracket notation

### LINKED ISSUE #42877: Set up passport challenge should allow for bracket notation
The challenge has a regex that only allows for dot notation.

`/secret:( |)process.env.SESSION_SECRET/gi`

Forum post: https://forum.freecodecamp.org/t/not-able-to-pass-session-and-session-secret-should-be-correctly-set-up/466681

Challenge: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-passport.md

### PR DESCRIPTION
**Pull request by alirezaghey**
Checklist:
- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes  #42877

fix minor bug in regex referring to literal dots with a dot

### CODE REVIEW

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-passport.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ Session and session secret should be correctly set up.
     (data) => {
       assert.match(
         data,
-        /secret:( |)process.env.SESSION_SECRET/gi,
+        /secret:( |)process\.env(\.SESSION_SECRET|\["SESSION_SECRET"\])/gi,
```
Comment (awu43) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-passport.md:114]:
```suggestion
        /secret:( |)process\.env(\.SESSION_SECRET|\[(?<q>"|')SESSION_SECRET\k<q>\])/gi,
```

Single quotes are also valid.

**Review by gikf — DISMISSED** [review]
LGTM :+1: 

I've tested all three possible options.

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-passport.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ Session and session secret should be correctly set up.
     (data) => {
       assert.match(
         data,
-        /secret:( |)process.env.SESSION_SECRET/gi,
+        /secret:( |)process\.env(\.SESSION_SECRET|\[(?<q>"|')SESSION_SECRET\k<q>\])/gi,
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-passport.md:114]:
```suggestion
        /secret:( |)process\.env(\.SESSION_SECRET|\[(?<q>"|')SESSION_SECRET\k<q>\])/g,
```

variables and method names are case sensitive, the `i` flag would pass stuff that doesn't work

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

**Review by majestic-owl448 — APPROVED** [review]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Review by gikf — APPROVED** [review]

**Pull request merged**